gadget/device_linux.go

```diff
@@ -211,13 +211,17 @@ func findDeviceForWritable() (device string, err error) {
 }
 
 func findParentDeviceWithWritableFallback() (string, error) {
-	deviceWritable, err := findDeviceForWritable()
+	partitionWritable, err := findDeviceForWritable()
 	if err != nil {
 		return "", err
 	}
+	return DiskFromPartition(partitionWritable)
+}
 
+// XXX: move to osutil?
+func DiskFromPartition(partition string) (string, error) {
```

Comment:
Maybe it should be named like `ParentDiskFromPartition`, because the implementation assumes that actual block device we want is one level higher in the hierarchy, which isn't true for anything that goes through device mapper (eg. LUKS, LVM and so on).

Reply:
Will the implementation that @cmatsuoka did for this in 7762 DTRT when it comes to LUKS/LVM ? If so it sounds like the followup for this should replace the implementation with the most robust one.

Reply:
Looked at the code in #7762. I think it should be fine as long as the partition in question is not under device mapper. We may want to add extra checks there detect such devices and error out.

Reply:
A block device under the device mapper could actually be in more than one physical disk (and partitioning the "parent disk" for a mapped device could not make sense in all cases), so I believe it's fine to assume that we're dealing with partitions on a physical disk instead of encrypted or logical volumes. In our current scenario we'll be always dealing with real disk partitions.